components/BarcodeScanner.vue

```diff
@@ -0,0 +1,334 @@
+            value: content,
+          };
+        }
+      });
+      return dlInfo;
+    },
+    cvsDrawArea() {
+      this.maskCanvas.width = this.clientWidth;
+      this.maskCanvas.height = this.clientHeight;
+      this.maskCanvasContext.fillStyle = 'rgba(0,0,0,.5)';
+      this.maskCanvasContext.fillRect(
+        0,
+        0,
+        this.maskCanvas.width,
+        this.maskCanvas.height
+      );
+      this.maskCanvasContext.globalCompositeOperation = 'destination-out';
+      this.maskCanvasContext.fillStyle = 'white';
+      this.maskCanvasContext.fillRect(
+        (this.regionLeft / 100) * this.maskCanvas.width,
+        (this.regionTop / 100) * this.maskCanvas.height,
+        ((100 - 2 * this.regionLeft) / 100) * this.maskCanvas.width,
+        ((100 - 2 * this.regionTop) / 100) * this.maskCanvas.height
+      );
+      this.maskCanvasContext.globalCompositeOperation = 'source-over';
+      this.maskCanvasContext.lineWidth = 2;
+      this.maskCanvasContext.strokeStyle = this.guideColor;
+      this.maskCanvasContext.rect(
+        (this.regionLeft / 100) * this.maskCanvas.width,
+        (this.regionTop / 100) * this.maskCanvas.height,
+        ((100 - 2 * this.regionLeft) / 100) * this.maskCanvas.width,
+        ((100 - 2 * this.regionTop) / 100) * this.maskCanvas.height
+      );
+      this.maskCanvasContext.stroke();
+    },
+    async updateCamera(deviceId) {
+      if(this.currentCamera.deviceId === deviceId) {
+        return;
+      }
+      this.loading = true;
+      await this.scanner.setCurrentCamera(deviceId);
+      this.currentCamera = await this.scanner.getCurrentCamera();
+      this.loading = false;
+    },
+    async upload(event) {
+      await this.imageScan();
+
+      const files = event.target.files;
+      // no file selected
+      if(files.length === 0) {
+        return;
+      }
+      for(let i = 0; i < files.length; ++i) {
+        const file = files[i];
+        const reader = new FileReader();
+        reader.onload = async () => {
+          const st = Date.now();
+          try {
+            this.scanning = true;
+            const results = await this.reader.decode(file);
+            if(results.length === 0) {
+              const message = 'No results found.';
+              this.$emit('error', message);
+              return;
+            }
+            this.$emit('result', this.getDLInfo(results[0].barcodeText));
+          } catch(e) {
+            console.error(e);
+            const message = 'There was an error scanning your license. ' +
+              'Please try again.';
+            this.$emit('error', message);
+          } finally {
+            const et = Date.now();
+            console.log('scan time', (et - st) + 'ms');
+            this.scanning = false;
+          }
+        };
+        reader.onerror = async e => {
+          console.error(e);
+          this.$emit('error', e.message);
+        };
+        reader.readAsDataURL(files[0]);
```

Comment:
This loops through all the files but always passes just the first one. Is this what was desired?

```suggestion
        reader.readAsDataURL(files[i]);
```